caikit/runtime/client/remote_module_base.py

```diff
@@ -0,0 +1,496 @@
+
+                # If the param is different then the default then add it to parameters
+                elif value != method.signature.default_parameters.get(param):
+                    http_request_dict[OPTIONAL_INPUTS_KEY][param] = value
+
+            # If there is only one input then collapse down the value
+            if len(http_request_dict[REQUIRED_INPUTS_KEY]) == 1:
+                http_request_dict[REQUIRED_INPUTS_KEY] = list(
+                    http_request_dict[REQUIRED_INPUTS_KEY].values()
+                )[0]
+        elif service_type == ServiceType.TRAINING:
+            # Strip all null values
+            def _remove_null_values(_attr):
+                if isinstance(_attr, dict):
+                    return {
+                        key: _remove_null_values(value)
+                        for key, value in _attr.items()
+                        if value
+                    }
+                if isinstance(_attr, list):
+                    return [
+                        _remove_null_values(listitem) for listitem in _attr if listitem
+                    ]
+
+                return _attr
+
+            http_request_dict = _remove_null_values(request_dm_dict)
+
+        # Get request options and target
+        request_kwargs = {"headers": {"Content-type": "application/json"}}
+        if self._tls.enabled:
+            if self._tls.cert_file and self._tls.key_file:
+                request_kwargs["cert"] = (
+                    self._tls.cert_file,
+                    self._tls.key_file,
+                )
+
+            if self._tls.insecure_verify:
+                request_kwargs["verify"] = False
+            else:
+                request_kwargs["verify"] = self._tls.ca_file or True
+
+        if method.output_streaming:
+            request_kwargs["stream"] = True
+
+        if self._connection.timeout:
+            request_kwargs["timeout"] = self._connection.timeout
+
+        request_kwargs = merge_configs(
+            request_kwargs, overrides=self._connection.options
+        )
+
+        request_url = (
+            f"{self._get_remote_target()}{get_http_route_name(method.rpc_name)}"
+        )
+
+        # Send request
+        response = requests.post(request_url, json=http_request_dict, **request_kwargs)
+        if response.status_code != 200:
+            raise CaikitCoreException(
+                CaikitCoreStatusCode.UNKNOWN,
+                f"Received status {response.status_code} from remote server: {response.text}",
+            )
+
+        # Parse response data model either as file or json
+        response_dm_class = DataBase.get_class_for_name(method.response_dm_name)
+
+        if method.output_streaming:
+
+            def stream_parser():
+                """Helper Generator to parse SSE events"""
+                for line in response.iter_lines():
+                    # Skip empty or event lines as they're constant
+                    if not line or b"event" in line:
```

Comment:
NIT: Do we need to do `line = line.strip()`? Probably safe to assume that lines are pre-stripped, but you never know?

Reply:
Alternately, should we just invert this and only `yield` on `if b"data: " in line`?

Reply:
I think inverting this is the real solution! Good call